expiringdict/__init__.py

```diff
@@ -69,7 +69,10 @@ def __setitem__(self, key, value):
         """ Set d[key] to value. """
         with self.lock:
             if len(self) == self.max_len:
-                self.popitem(last=False)
+                try:
+                    self.popitem(last=False)
```

Comment:
I assume that python 3.x raises exception if it is empty? If so then it should be a specific exception type so please catch only that specific exception in the `except` clause below.